src/main/java/LinkedBinarySearchTree.java

```diff
@@ -47,8 +46,8 @@ private Node<T> add(T element, Node<T> current) {
     public boolean remove(T element) {
         T returnElement = null;
         if (isEmpty()) {
-            throw new NoSuchElementException();
-        } else if (root.getData().equals(element)) {
+            throw new NoSuchElementException("Empty Tree");
+        } else if (root.getData().compareTo(element) == 0) {
             returnElement = root.getData();
             root = findReplacementNode(root);
         } else if (root.getData().compareTo(element) > 0) {
```

Comment:
Minor, can optimize a bit here now by pulling out `int cmp = root.getData().compareTo(element)`. Applies to a lot of methods now.

Reply:
In addition to pulling the comparison out of the `if` statements where it made sense, I also changed some `else if` to just `if` for consistency.